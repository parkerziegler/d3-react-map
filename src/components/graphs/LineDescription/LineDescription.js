import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { TransitionGroup, CSSTransition } from 'react-transition-group';

import '../../../stylesheets/LineDescription.scss';

export const LineDescription = ({ title, subtitle, description }) => (
  <TransitionGroup component={null}>
    <CSSTransition
      appear
      classNames="line-description-transition"
      timeout={3000}
    >
      <div className="line-description-container">
        <div className="line-description-title">
          <div className="title">{title}</div>
          <div className="subtitle">{subtitle}</div>
        </div>
        {description}
      </div>
    </CSSTransition>
  </TransitionGroup>
);

LineDescription.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

const mapStateToProps = (state) => ({
  title: state.router.result.descTitle,
  subtitle: state.router.result.descSubtitle,
  description: state.router.result.jsx,
});

export default connect(mapStateToProps)(LineDescription);
