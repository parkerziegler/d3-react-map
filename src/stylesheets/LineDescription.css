/** A constants SCSS partial for storing common variables and mixins */
@import url('https://fonts.googleapis.com/css?family=Roboto');
.line-description-container {
  display: flex;
  flex-direction: row;
  text-align: left;
}
.line-description-container .line-description-title {
  display: flex;
  flex-direction: column;
  height: 100%;
  min-width: 30%;
}
.line-description-container .line-description-title .title {
  font-size: 16px;
  color: #000000;
  font-family: 'HelveticaNeue', 'Helvetica', sans-serif;
  font-weight: bold;
  margin-bottom: 3px;
}
@media screen and (min-width: 480px) {
  .line-description-container .line-description-title .title {
    font-size: 18px;
  }
}
@media screen and (min-width: 768px) {
  .line-description-container .line-description-title .title {
    font-size: 22px;
  }
}
@media screen and (min-width: 1024px) {
  .line-description-container .line-description-title .title {
    font-size: 28px;
  }
}
.line-description-container .line-description-title .subtitle {
  font-size: 12px;
  color: #666666;
  font-family: 'HelveticaNeue', 'Helvetica', sans-serif;
}
@media screen and (min-width: 480px) {
  .line-description-container .line-description-title .subtitle {
    font-size: 14px;
  }
}
@media screen and (min-width: 768px) {
  .line-description-container .line-description-title .subtitle {
    font-size: 18px;
  }
}
@media screen and (min-width: 1024px) {
  .line-description-container .line-description-title .subtitle {
    font-size: 21px;
  }
}
.line-description-container .text {
  color: #000000;
  font: 14px 'HelveticaNeue', 'Helvetica', sans-serif;
  margin-bottom: 10px;
}
