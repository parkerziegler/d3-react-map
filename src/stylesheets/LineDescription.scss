@import './_constants.scss';

.line-description-transition-appear {
  opacity: 0;
}

.line-description-transition-appear.line-description-transition-appear-active {
  opacity: 1;
  transition: opacity 1s ease-in;
}

.line-description-container {
  @include flex-column;
  text-align: left;

  @media (min-width: map-get($breakpoints, 'tablet')) {
    flex-direction: row;
  }

  .line-description-title {
    @include flex-column;
    height: 100%;
    min-width: 30%;
    margin-bottom: 10px;

    @media (min-width: map-get($breakpoints, 'tablet')) {
      margin-bottom: 0;
      margin-right: 10px;
    }

    .title {
      @include font-size($font-sizes);
      color: $black;
      font-family: $helvetica;
      font-weight: bold;
      margin-top: 0;
      margin-bottom: 5px;
    }

    .subtitle {
      @include font-size($subheader-font-sizes);
      color: #666666;
      font-family: $helvetica;
      margin: 0;
    }
  }

  .text {
    color: $black;
    font-family: $helvetica;
    font-size: 14px;
    margin-top: 0;
    margin-bottom: 10px;
  }
}
