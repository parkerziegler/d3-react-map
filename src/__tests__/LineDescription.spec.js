import { LineDescription } from '../components/graphs/LineDescription/LineDescription';

describe('<LineDescription />', () => {
  const props = {
    title: 'Title',
    subtitle: 'Subtitle',
    description: 'Description',
  };

  it('renders without crashing', () => {
    const wrapper = shallow(<LineDescription {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
