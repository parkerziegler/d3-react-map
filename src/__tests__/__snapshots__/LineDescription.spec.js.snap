// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LineDescription /> renders without crashing 1`] = `
<TransitionGroup
  childFactory={[Function]}
  component={null}
>
  <CSSTransition
    appear={true}
    classNames="line-description-transition"
    timeout={3000}
  >
    <div
      className="line-description-container"
    >
      <div
        className="line-description-title"
      >
        <h1
          className="title"
        >
          Title
        </h1>
        <p
          className="subtitle"
        >
          Subtitle
        </p>
      </div>
      Description
    </div>
  </CSSTransition>
</TransitionGroup>
`;
